ledger/participant-integration-api/src/main/scala/platform/store/backend/common/UserManagementStorageBackendTemplate.scala

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.backend.common
+
+import java.sql.Connection
+
+import anorm.SqlParser.{int, str}
+import anorm.{RowParser, SqlParser, SqlStringInterpolation, ~}
+import com.daml.ledger.api.domain
+import com.daml.ledger.api.domain.UserRight
+import com.daml.ledger.api.domain.UserRight.{CanActAs, CanReadAs, ParticipantAdmin}
+import com.daml.ledger.api.v1.admin.user_management_service.Right
+import com.daml.lf.data.Ref
+import com.daml.lf.data.Ref.UserId
+import com.daml.platform.store.SimpleSqlAsVectorOf.SimpleSqlAsVectorOf
+import com.daml.platform.store.backend.UserManagementStorageBackend
+
+import scala.util.Try
+
+object UserManagementStorageBackendTemplate extends UserManagementStorageBackend {
+
+  /** Marker value for absence of party in a db table (`primary_party` and `for_party` columns).
+    *
+    * Oracle doesn't distinguish between empty strings and NULLs.
+    * If we used NULL/empty string in Oracle then in Postgres we would have to:
+    * a) If chose to use NULL value in Postgres then we would have to use partial index to ensure multi-column uniqueness constrains (`UNIQUE (user_internal_id, user_right, for_party)` index).
+    * b) If chose to use empty string value in Postgres, we would end up with DB aware backends having ot use 1) `... is NULL` in Oracle vs. 2) `... = ''` in Postgres
```

Comment:
```suggestion
    * b) use empty string value in Postgres; it would imply ending up with DB aware backends having ot use 1) `... is NULL` in Oracle vs. 2) `... = ''` in Postgres
```